buildtest/modules/spider.py

```diff
@@ -0,0 +1,103 @@
+import json
+import os
+import subprocess
+
+from buildtest.tools.defaults import BUILDTEST_SPIDER_FILE, BUILDTEST_CONFIG_FILE
+from buildtest.tools.config import load_configuration
+
+
+class Spider:
+    """Class declaration of Spider class"""
+
+    def __init__(self, tree=None):
+        """Initialize method for Spider class.
+
+        :param tree: User can specify a module tree to query from spider.
+        :type tree: str
+        """
+
+        #  if user specifies a tree, then run spider command otherwise read from configuration file.
+        if tree:
+            self.tree = tree
+            spider_cmd = f"{os.getenv('LMOD_DIR')}/spider -o spider-json {self.tree}"
+
+            out = subprocess.check_output(spider_cmd, shell=True).decode("utf-8")
+            self.spider_content = json.loads(out)
+
+        else:
+            with open(BUILDTEST_SPIDER_FILE, "r") as fd:
```

Comment:
This assumes that the file exists, which I don't think we can safely do.

Reply:
i was not sure about the ``else`` logic. We could simply use `MODULEPATH` instead of this file.